### PR TITLE
chore: Implemented json offer integration feature.

### PR DESCRIPTION
**Pull request by phoenix0113**
- Implemented the `json offerCreateContingentPool` integration feature.
- Implemented the `json offerAddLiquidity` integration feature.
- Fixed the issue when `yarn lint` and `yarn test` to deploy using GitHub action.
- Added `chainId` and `verifyingContract` to database

### CODE REVIEW

File: migrations/1604516429083-InitialTables.ts [hunk 1]
```diff
@@ -3,13 +3,102 @@ import { MigrationInterface, QueryRunner } from 'typeorm';
...
+                "permissioned_token" character varying NOT NULL,
+                "salt" character varying NOT NULL,
+                "signature" character varying NOT NULL,
+                "chain_id" integer,
+                "verifying_contract" character varying NOT NULL,
+                CONSTRAINT "PK_9dwmad8elvup9wognw95fhqa48h" PRIMARY KEY ("offer_hash"))`,
         );
         await queryRunner.query(
-            `CREATE TABLE "kv_store" ("key" character varying NOT NULL, "value" character varying, "created_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), CONSTRAINT "PK_a9c11644aa565bf7675e6bd23ef" PRIMARY KEY ("key"))`,
+            `CREATE TABLE IF NOT EXISTS "offer_add_liquidities" (
```
Comment (Walodja1987) [inline migrations/1604516429083-InitialTables.ts]: nit: can we rename to "offer_add_liquidity" please, i.e. use singular and not plural?

File: migrations/1604516429083-InitialTables.ts [hunk 2]
```diff
@@ -3,13 +3,102 @@ import { MigrationInterface, QueryRunner } from 'typeorm';
...
+                "taker" character varying NOT NULL,
+                "maker_collateral_amount" character varying NOT NULL,
+                "taker_collateral_amount" character varying NOT NULL,
+                "maker_direction" character varying NOT NULL,
+                "offer_expiry" character varying NOT NULL,
+                "minimum_taker_fill_amount" character varying NOT NULL,
+                "pool_id" character varying NOT NULL,
+                "salt" character varying NOT NULL,
+                "signature" character varying NOT NULL,
+                "chain_id" integer,
```
Comment (Walodja1987) [inline migrations/1604516429083-InitialTables.ts]: is NOT NULL needed here? Chain Id should never be missing

File: src/schemas/index.ts [hunk 3]
```diff
@@ -7,6 +7,8 @@ import * as rfqMakerConfigSchema from './rfq_maker_config.json';
 import * as rfqMakerConfigListSchema from './rfq_maker_config_list.json';
 import * as slippageModelFileSchema from './slippage_model_file_schema.json';
 import * as slippageModelSchema from './slippage_model_schema.json';
+import * as sraofferAddLiquiditySchema from './sra_offer_add_liquidity_schema.json';
```
Comment (Walodja1987) [inline src/schemas/index.ts]: use capital "O" after "sra" in line with line 11

File: src/schemas/sra_offer_add_liquidity_schema.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
+{
+    "id": "/sraofferAddLiquiditySchema",
```
Comment (Walodja1987) [inline src/schemas/sra_offer_add_liquidity_schema.json]: same here, use capital "O" after "sra"

File: src/utils/order_utils.ts [hunk 5]
```diff
@@ -4,8 +4,15 @@ import { BigNumber } from '@0x/utils';
...
 import { NULL_ADDRESS, ONE_SECOND_MS } from '../constants';
-import { PersistentSignedOrderV4Entity, SignedOrderV4Entity } from '../entities';
 import {
+    OfferAddLiquidityEntity,
+    OfferCreateContingentPoolEntity,
+    PersistentSignedOrderV4Entity,
+    SignedOrderV4Entity,
+} from '../entities';
+import {
+    LimitOfferAddLiquidity,
```
Comment (Walodja1987) [inline src/utils/order_utils.ts]: Is there a need to use "Limit" here? I think that's 0x order specific terminology. We can also leave it if it simplifies things for you. Not hung up on that one

File: src/utils/order_utils.ts [hunk 6]
```diff
@@ -105,6 +112,54 @@ export const orderUtils = {
         };
         return signedOrder;
     },
+    deserializeOfferCreateContingentPool: (
+        offerCreateContingentPoolEntity: OfferCreateContingentPoolEntity,
+    ): LimitOfferCreateContingentPool => {
```
Comment (Walodja1987) [inline src/utils/order_utils.ts]: See comment on "Limit" above

File: src/utils/order_utils.ts [hunk 7]
```diff
@@ -105,6 +112,54 @@ export const orderUtils = {
...
+        const signedOffer: LimitOfferCreateContingentPool = {
```
Comment (Walodja1987) [inline src/utils/order_utils.ts]: here as well...

File: src/utils/order_utils.ts [hunk 8]
```diff
@@ -105,6 +112,54 @@ export const orderUtils = {
...
+            capacity: offerCreateContingentPoolEntity.capacity as string,
+            permissionedERC721Token: offerCreateContingentPoolEntity.permissionedERC721Token as string,
+            salt: offerCreateContingentPoolEntity.salt as string,
+            signature: JSON.parse(offerCreateContingentPoolEntity.signature as string),
+            chainId: Number(offerCreateContingentPoolEntity.chainId),
+            verifyingContract: offerCreateContingentPoolEntity.verifyingContract as string,
+        };
+        return signedOffer;
+    },
+    deserializeOfferAddLiquidity: (offerAddLiquidityEntity: OfferAddLiquidityEntity): LimitOfferAddLiquidity => {
```
Comment (Walodja1987) [inline src/utils/order_utils.ts]: see comment on "Limit" above

File: docker-compose.yml [hunk 9]
```diff
@@ -1,22 +1,25 @@
...
-            - ./postgres:/var/lib/postgresql/data
-        ports:
-            - '5432:5432'
\ No newline at end of file
+  postgres:
+    container_name: postgres_container
+    image: postgres
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
```
Comment (Walodja1987) [inline docker-compose.yml]: Is there a need to push this to github?

File: migrations/1604516429083-InitialTables.ts (hunk 2 above)
Comment (phoenix0113) [reply migrations/1604516429083-InitialTables.ts:66]: Okay. I will add `NOT NULL`.

File: src/schemas/index.ts (hunk 3 above)
Comment (phoenix0113) [reply src/schemas/index.ts:10]: Sorry. I will fix this issue.

File: src/schemas/sra_offer_add_liquidity_schema.json (hunk 4 above)
Comment (phoenix0113) [reply src/schemas/sra_offer_add_liquidity_schema.json:2]: I will fix this issue.

File: src/utils/order_utils.ts (hunk 7 above)
Comment (phoenix0113) [reply src/utils/order_utils.ts:118]: I will fix it.

File: src/utils/order_utils.ts (hunk 6 above)
Comment (phoenix0113) [reply src/utils/order_utils.ts:117]: I will fix it.

File: src/utils/order_utils.ts (hunk 5 above)
Comment (phoenix0113) [reply src/utils/order_utils.ts:14]: I will fix it.

File: docker-compose.yml (hunk 9 above)
Comment (phoenix0113) [reply docker-compose.yml:9]: I will add this information to .env file.

**Pull request merged**